src/components/results/result-list/result-list.component.ts

```diff
@@ -119,7 +113,7 @@ export class ResultListComponent implements OnInit, DoCheck, OnChanges {
    * @Input : Angular
    * @description An input to customize the resultlist
    */
-  @Input() public options: ResultListOptions = this.defaultOptions;
+  @Input() public options: ResultListOptions = new ResultListOptions();
```

Comment:
If input object doesn't contains all keys, there is no overriding with default value